src/GeolocationContext.tsx

```diff
@@ -258,14 +257,12 @@ export const GeolocationContextProvider: React.FC = ({children}) => {
 
         if (state.status != status) {
           dispatch({type: 'PERMISSION_CHANGED', status, locationEnabled});
-          if (enable_intercom) {
-            await updateChatUserMetadata({'AtB-App-Location-Status': status});
-          }
+          updateMetadata({'AtB-App-Location-Status': status});
         }
       }
     }
     checkPermission();
-  }, [appStatus, state.status, enable_intercom]);
+  }, [appStatus, state.status, updateMetadata]);
```

Comment:
Looks good in general, my only concern might be to have this function in the dependencies, it might create some complications

Reply:
any suggestion how to deal with that dependency? 🤔

Reply:
I remember @gorandalum and I did a trick to remove it from dependencies, perhaps hr could refresh my memory here or if it is fine to do in that way.

Reply:
Depends on what we want. Do we want the `checkPermission` function to rerun when the `enable_intercom`/`updateMetadata` is changed? Should some effects run when it is changed, and some not? Deciding this will a hint on the best way forward.

In general if we wan't to use something, but it shouldn't trigger rerenders, then it should be refs instead of state.